lib/rspec/core/example_group.rb

```diff
@@ -557,7 +557,9 @@ def self.run_examples(reporter)
           instance = new(example.inspect_output)
           set_ivars(instance, before_context_ivars)
           succeeded = example.run(instance, reporter)
-          RSpec.world.wants_to_quit = true if fail_fast? && !succeeded
+          if !succeeded && fail_fast?
+            RSpec.world.wants_to_quit = true if failures_number_met?(reporter.failed_examples.count)
+          end
```

Comment:
It seems odd to do this:

``` ruby
if some_condition
  do_something if some_other_condition
end
```

...rather than:

``` ruby
if some_condition && some_other_condtion
  do_something
end
```

Mind refactoring to the latter?